### ISSUE TITLE
A url list populated in database

### ISSUE THREAD

**Issue by jas-deep** [gophish/gophish#854]
How about having a list of urls (domains) (that I have got) populated in the database and i can pick the one i want to use for each campaign. I feel it will be useful to have them in the tool at one place.

UI to provide option to add/edit/delete the url list.

**Comment by jordan-wright** [comment]
I'd be interested to hear how many URLs people actually use. It seems like there'd be only a few which to me wouldn't warrant an entire API/UI to add/edit/delete.

I've been wrong before, though, so let's get some feedback and figure out the best way forward 😄

**Comment by S0larflare** [comment]
I use quite a number, but that’s due to me having everything internal and relying on DNS cname trickery

Sent from my iPhone

On 24 Nov 2017, at 16:11, Jordan Wright <notifications@github.com<mailto:notifications@github.com>> wrote:


I'd be interested to hear how many URLs people actually use. It seems like there'd be only a few which to me wouldn't warrant an entire API/UI to add/edit/delete.

I've been wrong before, though, so let's get some feedback and figure out the best way forward 😄

**Comment by hatsaresilly** [comment]
I just recently started using gophish, and think it's great. I currently only have one URL, but have plans for more. However, I'd also like to point out that even if you only have one URL, it would keep you from making a mistake and would be very convenient to select it from a drop down especially if running many campaigns. I would like to see this feature implemented eventually.

**Comment by skim-milk** [comment]
Just saw this from #349.
We're after that feature too, since we may run multiple campaigns for an organisation. E.g. training and awareness, OWA, password reset etc. These are generally located at different domains that we utilise.

Any ideas how to implement this in the config file going forward? I might give it a crack.

**Comment by pat-inthe-hat** [comment]
I agree that having the URLs populated in the UI would be a handy feature as I use between 5-10 domains.

**Comment by azymeri** [comment]
I'm actualy using ca. 20 LandingPages and more than two are active at same time. At the moment i need to run many gophish instances on diferent Ports beacuse of SSL.

**Comment by russklvacek** [comment]
+1 to the idea of being able to serve multiple SSL instances/domains
 from the same, single GoPhish installation.

**Comment by ad0nis** [comment]
+1 on this as well. Team is in transition from king-phisher to go-phish. It could manage many simultaneous SE campaigns on different domains with different certs for each domain, so we feel like we lost an important feature here. In king-phisher, similar to apache or nginx, it's more-or-less just a question of setting up the default cert that they'll see if they connect to your IP, and then the rest are used when a request comes in with the appropriate vhost in the SNI...

**Comment by glennzw** [comment]
I don't know if it helps, but this is my custom setup:
 * List of domains I own in a json blob
 * When creating new campaign the phishing URL parameter is pseudo randomly generated like so:
   * `https://<pseudo-random-words>.<random-domain>/<random-endpoints>`

On my server I have wildcard domains setup for my domains, so from the above I might get something like:

`https://server.foo.mydomain.com/users/update?rid/1234567`
